File: src/models/Product.js
const mongoose = require("mongoose");

const productSchema = new mongoose.Schema(
  {
    shopId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Shop",
      required: true,
      index: true
    },
    name: {
      type: String,
      required: true,
      trim: true
    },
    description: {
      type: String,
      trim: true,
      default: ""
    },
    price: {
      type: Number,
      required: true,
      min: 0
    },
    imageUrl: {
      type: String,
      trim: true,
      default: ""
    },
    category: {
      type: String,
      trim: true,
      default: "General"
    },
    isAvailable: {
      type: Boolean,
      default: true
    },
    sortOrder: {
      type: Number,
      default: 0
    }
  },
  {
    timestamps: true
  }
);

productSchema.index({ shopId: 1, isAvailable: 1 });
productSchema.index({ shopId: 1, category: 1 });

module.exports = mongoose.model("Product", productSchema);

File: owner_app/lib/src/models/product.dart
import '../core/utils/json_utils.dart';

class Product {
  const Product({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.imageUrl,
    required this.category,
    required this.isAvailable,
    required this.sortOrder,
  });

  final String id;
  final String name;
  final String description;
  final double price;
  final String imageUrl;
  final String category;
  final bool isAvailable;
  final int sortOrder;

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: stringFrom(json['id'] ?? json['_id']),
      name: stringFrom(json['name']),
      description: stringFrom(json['description']),
      price: doubleFrom(json['price']),
      imageUrl: stringFrom(json['imageUrl']),
      category: stringFrom(json['category'], fallback: 'General'),
      isAvailable: boolFrom(json['isAvailable'], fallback: true),
      sortOrder: doubleFrom(json['sortOrder']).toInt(),
    );
  }
}

